packages/@eventual/aws-runtime/src/handlers/apig-command-adapter.ts

```diff
@@ -21,24 +23,28 @@ export function createApiGCommandAdaptor({
   commandWorker,
   serviceClientBuilder,
   serviceSpec,
+  serviceName: _serviceName,
 }: {
   commandWorker: CommandWorker;
+  serviceName: LazyValue<string>;
   serviceSpec?: ServiceSpec;
   serviceClientBuilder?: (serviceUrl: string) => EventualServiceClient;
 }): APIGatewayProxyHandlerV2 {
+  const serviceName = getLazy(_serviceName);
```

Comment:
Should we try and get it from the header?

Reply:
The url is coming from the headed. the service name is already in the envelope variables